src/track/beats.h

```diff
@@ -37,18 +37,20 @@ namespace mixxx {
 class Beats final : public QObject {
     Q_OBJECT
   public:
-    /// Construct a BeatMap. iSampleRate may be provided if a more accurate
-    /// sample rate is known than the one associated with the Track.
+    /// iSampleRate may be provided if a more accurate sample rate is known
+    /// than the one associated with the Track.
     explicit Beats(const Track* track, SINT iSampleRate = 0);
-    /// Construct a BeatMap. iSampleRate may be provided if a more accurate
-    /// sample rate is known than the one associated with the Track. If it is
-    /// zero then the track's sample rate will be used. The BeatMap will be
-    /// deserialized from the byte array.
+    /// iSampleRate may be provided if a more accurate sample rate is known
+    /// than the one associated with the Track. If it is zero then the track's
+    /// sample rate will be used.
```

Comment:
Use std::optional rather than magic numeric values.

Reply:
Nevermind, I see this is obsoleted by the next commit.

Reply:
Thanks for noticing this, it was bugging me a lot as well. There are two options, either introduce a bool member variable to the Frame class which indicates it is an invalid frame or use std::optional which is more suitable for non-deterministic search operations like these.